test/select.spec.js

```diff
@@ -399,6 +399,23 @@ describe('ui-select tests', function() {
     expect(getMatchLabel(el)).toEqual('Samantha');
   });
 
+  it('should correctly render initial state with track by $index', function () {
+    
+    var el = compileTemplate(
+      '<ui-select ng-model="selection.selected"> \
+        <ui-select-match placeholder="Pick one...">{{$select.selected.name}}</ui-select-match> \
+        <ui-select-choices repeat="person in people track by $index"> \
+          {{person.email}} \
+        </ui-select-choices> \
+      </ui-select>'
+    );
+
+    openDropdown(el);
+
+    var generatedId = el.scope().$select.generatedId;
+    expect($(el).find('[id$="row-' + generatedId + '-0"]').length).toEqual(1);

```

Comment:
Would this capture all elements if they shared that id? (In case a bug was introduced)

I haven't jQuery'd in years, according to http://stackoverflow.com/a/6744674 it wouldn't? I'll have to check later.